### PR TITLE
Introduce rbac flag to optionate role and rolebindings for namespace pools

### PR DESCRIPTION
**Pull request by Shubham-astro**
## Description

This PR introduces a keep flag so the already create role and rolebindings are not deleted and synchronize on a helm update/upgrade. Disable management of commander namespace resources.

## Related Issues
Related https://github.com/astronomer/issues/issues/6582

## Testing
Do not merge this PR until this text is replaced with details about how these changes were tested.

## Merging
Do not merge this PR until it lists which release branches this PR should be merged / cherry-picked into.

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -41,6 +41,7 @@ global:
     # namespace pools: define k8s namespaces in which Astronomer will manage Airflow deployments
     namespacePools:
       enabled: false
+      keeproleandrolebindings: false
```
Comment (pgvishnuram) [inline values.yaml:44]: it can just be role: true or false it should be default to true

Comment (pgvishnuram) [reply values.yaml:44]: here we are not keeping the existing roles we are creating an option to either allow creation of roles by platform or by customer and our naming

Comment (Shubham-astro) [reply values.yaml:44]: @pgvishnuram one of the acceptance criteria says: 
Introduce a keep flag so already create role and rolebindings are not deleted and synchronize on a helm update/upgrade.

Comment (pgvishnuram) [reply values.yaml:44]: why do we need a keep flag here rbac: true is default unless customer sets them to false so this is fully of customer responsibility to make revoke resources and the naming ```keeproleandrolebindings``` does make any for the customers who already has those rbac applied

Comment (Shubham-astro) [reply values.yaml:44]: done, since by default we are keeping the value to true so customer when needs to recreate role can mark it as false

**Comment by Shubham-astro** [comment]
@pgvishnuram can you re-review.?

File: charts/astronomer/templates/commander/commander-role.yaml [hunk 2]
```diff
@@ -5,6 +5,7 @@
 # 1. Do not create any resource if rbacEnabled is disabled
 # 2. Create a Cluster Role if clusterRoles and rbacEnabled are enabled and namespacePools is disabled
 # 3. Create roles for each namespaces in the namespacePool (+ astronomer namespace) if enabled.
+{{- $commanderrole:= and .Values.global.features.namespacePools.keeproleandrolebindings (not $.Values.global.features.namespacePools.namespaces.create)}}
 {{- $useClusterRoles := and .Values.global.rbacEnabled .Values.global.clusterRoles (not .Values.global.features.namespacePools.enabled)}}
 {{- $shouldCreateResources := and .Values.global.rbacEnabled (or .Values.global.clusterRoles .Values.global.features.namespacePools.enabled) }}
```
[the remark this answers is not in the document]
Comment (danielhoherd) [reply charts/astronomer/templates/commander/commander-role.yaml:10]: @pgvishnuram 

> both should be replaced by above condition

both what? can you be more specific?

[the remark this answers is not in the document]
Comment (Shubham-astro) [reply charts/astronomer/templates/commander/commander-role.yaml]: this is done now.

File: values.yaml [hunk 3]
```diff
@@ -41,6 +41,8 @@ global:
     # namespace pools: define k8s namespaces in which Astronomer will manage Airflow deployments
     namespacePools:
       enabled: false
+      # if rbacCreate is set to false then platform does not create roles and rolebinding.
+      rbacCreate: true
```
Comment (danielhoherd) [inline values.yaml:45]: This should probably be createRbac so it follows the same form as other variables we use, like `createJetStreamJob`, `createSecret`, `createDefaultIndex`, etc..

Comment (Shubham-astro) [reply values.yaml]: done, changes to createRbac

**Review by danielhoherd — APPROVED** [review]
Seems good

**Review by pgvishnuram — APPROVED** [review]

**Pull request merged**